### PR TITLE
DOC: Examples for special functions related to the student t distribution

### PR DESCRIPTION
**Pull request by dschmitz89**
#### Reference issue
gh-7168

#### What does this implement/fix?
Examples for special functions related to the student t distribution:
* `special.stdtr`
* `special.stdtrit`
* `special.stdtridf`

### CODE REVIEW

**Comment by dschmitz89** [comment]
@melissawm friendly ping: any chance that this can be merged? :)

**Comment by WarrenWeckesser** [comment]
@dschmitz89, I'm taking a look now.

For reviewer's convenience, here are the links to the updated docs:
* [`special.stdtr`](https://output.circle-artifacts.com/output/job/7eea9e94-99fe-44be-be0a-1b8d34bef2c5/artifacts/0/html/reference/generated/scipy.special.stdtr.html)
* [`special.stdtrit`](https://output.circle-artifacts.com/output/job/7eea9e94-99fe-44be-be0a-1b8d34bef2c5/artifacts/0/html/reference/generated/scipy.special.stdtrit.html)
* [`special.stdtridf`](https://output.circle-artifacts.com/output/job/7eea9e94-99fe-44be-be0a-1b8d34bef2c5/artifacts/0/html/reference/generated/scipy.special.stdtridf.html)

File: scipy/special/_add_newdocs.py [hunk 1]
```diff
@@ -12579,7 +12664,73 @@ def add_newdoc(name, doc):
     --------
     stdtr : Student t CDF
     stdtridf : inverse of stdtr with respect to `df`
+    scipy.stats.t : Student t distribution
+
+    Notes
+    -----
+    The student t distribution is also available as `scipy.stats.t`. Calling
+    `stdtrit` directly can improve performance compared to the ``ppdf``
```
Comment (WarrenWeckesser) [inline scipy/special/_add_newdocs.py:12672]:
```suggestion
    `stdtrit` directly can improve performance compared to the ``ppf``
```

File: scipy/special/_add_newdocs.py [hunk 2]
```diff
@@ -12579,7 +12664,73 @@ def add_newdoc(name, doc):
...
+    --------
+    `stdtrit` represents the inverse of the student t distribution CDF which
+    is available as `stdtr`. Here, we calculate the CDF for ``df`` at
+    ``x=1``. `stdtrit` then returns ``1`` up to floating point errors
+    given the same value for `df` and the computed CDF value.
+
+    >>> import numpy as np
+    >>> from scipy.special import stdtr, stdtrit
+    >>> import matplotlib.pyplot as plt
+    >>> df, x = 3, 1
```
Comment (WarrenWeckesser) [inline scipy/special/_add_newdocs.py:12685]: Style nit, here and elsewhere: multiple assignment like this is generally (IME) a bad practice.
```suggestion
    >>> df = 3
    >>> x = 1
```

**Review by WarrenWeckesser — CHANGES_REQUESTED** [review]
Looks good!

I made a few inline suggestions.  One more editorial change: in the `stdtr` docstring, let's move the comment about it being "the integral from minus infinity to t of the Student t distribution [sic]" to the notes, and fix the text to something simpler, such as "The function computes the integal: ...", because "the integral of the distribution" isn't quite right.  Also, move any terms in the integrand that do not depend on $t$ out of the integral.

**Review by tupui — APPROVED** [review]
LGTM and renders good. Thanks @dschmitz89

**Pull request merged**